src/ucan-chain.js

```diff
@@ -0,0 +1,219 @@
+import * as ucan from './ucan.js'
+
+/**
+ * @template A
+ * @param {import('./types').CapabilitySemantics<A>} semantics
+ * @param {UcanChain} ucan
+ */
+export function* findValidCaps(semantics, ucan) {
+  const caps = ucan.capabilities()
+  const parentCaps = []
+
+  for (const cap of caps) {
+    if (cap.with.startsWith('prf:')) {
+      const proofs = ucan.proofs()
+      const [, index] = cap.with.split(':')
+      const parsedIndex = Number.parseInt(index)
+      if (index === '*') {
+        for (const proof of proofs) {
+          parentCaps.push(...proof.capabilities())
```

Comment:
Does this not need to be recursive, so that if the capabilities of the parent proofs are also in the format `prf:X` then we also expand those out?

Reply:
👍